aws-api/src/main/java/com/amplifyframework/api/aws/MutiAuthSubscriptionOperation.java

```diff
@@ -116,7 +116,20 @@ private void dispatchRequest() {
                     MutiAuthSubscriptionOperation.this.subscriptionId = subscriptionId;
                     onSubscriptionStart.accept(subscriptionId);
                 },
-                onNextItem,
+                response -> {
+                    if (response.hasErrors()) {
+                        // If there are auth-related errors, dispatch an ApiAuthException
+                        if (hasAuthRelatedErrors(response)) {
+                            executorService.submit(this::dispatchRequest);
+                            return;
+                        }
+                        // Otherwise, we just want to dispatch it as a next item and
+                        // let callers deal with the errors.
+                        onNextItem.accept(response);
+                    } else {
+                        onNextItem.accept(response);
+                    }
```

Comment:
Same logic, just a little simplified - 

```suggestion
                    if (response.hasErrors() && hasAuthRelatedErrors(response)) {
                        // If there are auth-related errors, dispatch an ApiAuthException
                        executorService.submit(this::dispatchRequest);
                    } else {
                        // Otherwise, we just want to dispatch it as a next item and
                        // let callers deal with the errors.
                        onNextItem.accept(response);
                    }
```

Reply:
Good call on this one...Just committed this change as well.